ui/raidboss/emulator/data/network_log_converter/LineEvent0x00.ts

```diff
@@ -1,14 +1,24 @@
 import LineEvent from './LineEvent';
+import LogRepository from './LogRepository';
+
+export type LineEvent0x00Parts = [
+  string, // 0: ID
+  string, // 1: Timestamp
+  string, // 2: Type
+  string, // 3: Speaker
+  string, // 4: Message
+  string, // 5: Checksum
+];
```

Comment:
I think if you want to document what the various parts are here, I think there are stronger ways to do it where the names become identifiers rather than comments.

One option here is to make an "enum" (e.g. `const field = { id: 0, timestamp: 1, etc } as const;` and then some other code can say `parts[field.type]` instead of `parts[2]`).

Reply:
I don't know how that'll pan out with the issues outlined in the first review re: `as string` etc.

Reply:
Additionally, that's really not going to work for 0x15 and 0x16, with the offset stuff.

Reply:
What about writing like this:
```typescript
export type LineEvent0x00Parts = [
  id: string,
  timestamp: string,
  type: string,
  speaker: string,
  message: string,
  checksum: string,
];
```

So you could get a type hint as well.
![image](https://user-images.githubusercontent.com/19927330/119218802-e32c3f80-bb14-11eb-8319-c037bfba03c4.png)

Reply:
Given that many of these properties are needed regardless, maybe it makes sense to just flat out make these getters on the class, such that:

```typescript

// Chat event
export class LineEvent0x00 extends LineEvent {
  constructor(repo: LogRepository, line: string, public parts: LineEvent0x00Parts) {
    super(repo, line, parts);
    if (this.parts.slice(4).join('|').split('\u001f\u001f').length > 1)
      this.invalid = true;
  }

  public get id(): string {
    return this.parts[0];
  }

  public get timestamp(): string {
    return this.parts[1];
  }
...
```

Thoughts?

Reply:
I assume with nullish coalescing or something to handle missing parts as well?

```typescript
  public get id(): string {
    return this.parts[0] ?? '';
  }

  public get timestamp(): string {
    return this.parts[1] ?? '';
  }
  ```

That seems reasonable to me!

Reply:
Yeah, with nullish coalescing, and `parseInt`/`parseFloat` where appropriate.

Reply:
Given this adjustment, I'm going to pull `LineEvent.ts` into this PR to put `id` and `timestamp` on the base class, otherwise it'll cause issues and those fields will have to be adjusted again later.